lib/datasource/artifactory/index.ts

```diff
@@ -0,0 +1,97 @@
+import { logger } from '../../logger';
+import { ExternalHostError } from '../../types/errors/external-host-error';
+import { cache } from '../../util/cache/package/decorator';
+import { parse } from '../../util/html';
+import { HttpError } from '../../util/http/types';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { datasource, defaultRegistryUrl } from './common';
+
+export class ArtifactoryDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+
+  override readonly caching = true;
+
+  override readonly registryStrategy = 'merge';
+
+  @cache({
+    namespace: `datasource-${datasource}`,
+    key: ({ registryUrl }: GetReleasesConfig) => `${registryUrl}`,
+  })
+  async getReleases({
+    lookupName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    if (registryUrl === defaultRegistryUrl) {
+      logger.warn(
+        'artifactory datasource requires custom registryUrl. Skipping datasource'
+      );
+      return null;
+    }
+
+    const url = `${registryUrl}/${lookupName}`;
+
+    const result: ReleaseResult = {
+      homepage: 'https://jfrog.com/artifactory',
```

Comment:
homepage should be the home page for the package, not for Artifactory itself. Remove this line

Reply:
```suggestion
```

Reply:
Removed it.